crates/sui-framework/sources/governance/validator_set.move

```diff
@@ -513,18 +526,56 @@ module sui::validator_set {
     }
 
     /// Process all active validators' pending delegation deposits and withdraws.
-    fun process_pending_delegations_and_withdraws(
+    fun process_pending_delegations(
         validators: &mut vector<Validator>, new_epoch: u64, ctx: &mut TxContext
     ) {
         let length = vector::length(validators);
         let i = 0;
         while (i < length) {
             let validator = vector::borrow_mut(validators, i);
-            validator::process_pending_delegations_and_withdraws(validator, new_epoch, ctx);
+            validator::process_pending_delegations(validator, new_epoch, ctx);
             i = i + 1;
         }
     }
 
+    /// Go through the global withdraw queue, calculate the limit for each pool, and
+    /// process withdraw request up to that limit for each pool.
+    fun process_pending_delegation_withdraws(
+        self: &mut ValidatorSet, max_delegation_withdraw_per_epoch: u64, new_epoch: u64, ctx: &mut TxContext
+    ) {
+        // A temp table (address -> u64) that stores how many withdraw requests we are procssing for each
```

Comment:
If this is temporary, it's probably best not to use `linked_table`--will end up doing a lot of needless object allocations and hashing. I would suggest `vec_map`.